### ISSUE TITLE
 ifAlias script not polled right with snmpbulkwalk - 2

### ISSUE THREAD

**Issue by patrick7** [librenms/librenms#5547]
Hi,

It looks like the issue addressed in https://github.com/librenms/librenms/issues/2002 is still present. Please also see https://community.librenms.org/t/librenms-interface-issues/444

Observium's perl script is working. 

Regards
Patrick

**Comment by laf** [comment]
You submitted a pull request so I assumed you'd fixed the issue.

To start with provide the full output from running `./scripts/ifAlias` manually.

**Comment by patrick7** [comment]
The pull request fixed a problem with interfaces containing an "@" :-) 

There is still a problem that snmpbulkwalk (which is used by librenms) does not show the complete data. snmpwalk does.

Output:
```
(ZRH)root@monitor:/var/www/xyz.li/subdomains/librenms# ./scripts/ifAlias 
.1.3.6.1.2.1.31.1.1.1.18.0
noSuchName
```

From a remote system with snmpbulkwalk:
```
/usr/bin/snmpbulkwalk -v2c -c secret -OQUs -m IF-MIB -M /var/www/xyz.li/subdomains/librenms/mibs:/var/www/xyz.li/subdomains/librenms/mibs/supermicro 185.xx.xx.254 ifXEntry | grep ifAlias
ifAlias.1 = 
ifAlias.2 = 
ifAlias.3 = 
ifAlias.4 = 
ifAlias.5 = 
ifAlias.6 = 
ifAlias.7 = 
ifAlias.8 = 
ifAlias.9 = Transit: Upstream ISP
ifAlias.10 = Core: Loopback
ifAlias.11 = 
ifAlias.12 = 
ifAlias.15 = 
ifAlias.16 = Cust: Patrick IPv6 Home
ifAlias.17 = 
ifAlias.18 = 
ifAlias.19 = 
ifAlias.20 = Core: LACP vms1.xyz.li
ifAlias.22 = Server: Public
ifAlias.23 = Server: Private
ifAlias.24 = Server: Monitoring
ifAlias.25 = Cust: WLAN
ifAlias.26 = Server: RIPE Atlas
ifAlias.27 = Server: Access Point
ifAlias.28 = Core: Tunnel Frankfurt
ifAlias.29 = Core: Tunnel Home
ifAlias.30 = Server: Management
```

snmpwalk:
```
snmpwalk -v2c -c secret cr1.zrh1.xyz.li .1.3.6.1.2.1.31.1.1.1.18
IF-MIB::ifAlias.1 = STRING: 
IF-MIB::ifAlias.2 = STRING: Core: LACP vms1.xyz.li Port 1
IF-MIB::ifAlias.3 = STRING: Core: LACP vms1.xyz.li Port 2
IF-MIB::ifAlias.4 = STRING: Core: Bla
IF-MIB::ifAlias.5 = STRING: Core: x.probes.atlas.ripe.net
IF-MIB::ifAlias.6 = STRING: Core: ap1.zrh1.xyz.li
IF-MIB::ifAlias.7 = STRING: 
IF-MIB::ifAlias.8 = STRING: 
IF-MIB::ifAlias.9 = STRING: Transit: Upstream ISP
IF-MIB::ifAlias.10 = STRING: Core: Loopback
IF-MIB::ifAlias.11 = STRING: 
IF-MIB::ifAlias.12 = STRING: 
IF-MIB::ifAlias.15 = STRING: 
IF-MIB::ifAlias.16 = STRING: Cust: Patrick xyz IPv6 Home
IF-MIB::ifAlias.17 = STRING: 
IF-MIB::ifAlias.18 = STRING: 
IF-MIB::ifAlias.19 = STRING: 
IF-MIB::ifAlias.20 = STRING: Core: LACP vms1.xyz.li
IF-MIB::ifAlias.22 = STRING: Server: Public
IF-MIB::ifAlias.23 = STRING: Server: Private
IF-MIB::ifAlias.24 = STRING: Server: Monitoring
IF-MIB::ifAlias.25 = STRING: Cust: WLAN
IF-MIB::ifAlias.26 = STRING: Server: RIPE Atlas
IF-MIB::ifAlias.27 = STRING: Server: Access Point
IF-MIB::ifAlias.28 = STRING: Core: Tunnel Frankfurt
IF-MIB::ifAlias.29 = STRING: Core: Tunnel Home
IF-MIB::ifAlias.30 = STRING: Server: Management
```

**Comment by geordish** [comment]
I've created a PR to fudge a fix for this. Mind testing?

**Comment by patrick7** [comment]
Thanks for your messages. Just tested it, but it does show the same problem as before on my system (debian8).

**Comment by laf** [comment]
Please provide updated debug info

**Comment by geordish** [comment]
I'm running Debian 8.7, and it works perfectly fine for me... are you sure you copied the new ifAlias script onto your target, and are running it?

**Comment by patrick7** [comment]
Closed. Problem has been fixed with  #6005
